contracts/libs/Mappings.sol

```diff
@@ -0,0 +1,232 @@
+// SPDX-License-Identifier: MIT
+// heavily inspired by: https://bitbucket.org/rhitchens2/soliditystoragepatterns/src/master/GeneralizedCollection.sol
```

Comment:
I think you mean: https://bitbucket.org/rhitchens2/soliditystoragepatterns/src/master/OneToMany.sol
It does not have a license, so we cannot just use it.